pkg/apis/postgres-operator.crunchydata.com/v1beta1/postgrescluster_types.go

```diff
@@ -531,6 +531,44 @@ type PostgresVolumesSpec struct {
 	// ---
 	// +optional
 	Temp *VolumeClaimSpec `json:"temp,omitempty"`
+
+	// Additional pre-existing volumes to add to the pod.
+	// ---
+	// +optional
+	// +listType=map
+	// +listMapKey=name
+	// +kubebuilder:validation:MaxItems=10
+	Additional []*AdditionalVolume `json:"additional,omitempty"`
+}
+
+type AdditionalVolume struct {
+	// The name of the volume used for mounting path.
+	// Volumes are mounted in the pods at `volumes/<NAME>`
+	// Must be unique.
+	// ---
+	// The `Name` field is a `DNS1123Subdomain` type to enforce
+	// the max length and also allow us to more easily transition
+	// to CPK-provisioned volumes.
+	// +kubebuilder:validation:Required
+	Name DNS1123Subdomain `json:"name,omitempty"`
```

Comment:
⛏️ Let's use the k8s upstream marker for this.
🔧 Drop the `omitempty` on required fields.


```suggestion
	// +required
	Name DNS1123Subdomain `json:"name"`
```